jdbc/src/main/java/org/jclouds/jdbc/service/JdbcService.java

```diff
@@ -16,54 +16,243 @@
+      blobEntity.setKey(blob.getMetadata().getName());
+      blobEntity.setBlobAccess(blobAccess);
+      blobEntity.setEtag(DIRECTORY_MD5);
+      return blobRepository.create(blobEntity);
+   }
+
+   @Transactional
+   public BlobEntity createDirectoryBlob(String containerName, Blob blob) {
+      return createDirectoryBlob(containerName, blob, null);
+   }
+
+   @Transactional(rollbackOn = IOException.class)
+   public BlobEntity createOrModifyBlob(String containerName, Blob blob) throws IOException {
+      return createOrModifyBlob(containerName, blob, null);
+   }
+
+   @Transactional
+   public BlobEntity findBlobById(String containerName, String key) {
+      ContainerEntity containerEntity = containerRepository.findContainerByName(containerName);
+      return containerEntity == null ? null : blobRepository.find(new BlobEntityPK(containerEntity.getId(), key));
    }
 
    @Transactional
-   public void deleteContainerByName(String name) {
-      containerRepository.deleteContainerByName(name);
+   public ChunkEntity findChunkById(Long id) {
+      return chunkRepository.find(id);
    }
 
    @Transactional
-   public void setContainerAccessByName(String name, ContainerAccess access) {
-      Container container = containerRepository.findContainerByName(name);
-      container.setContainerAccess(access);
-      containerRepository.save(container);
+   public List<BlobEntity> findBlobsByContainer(String containerName) {
+      return blobRepository.findBlobsByContainer(containerRepository.findContainerByName(containerName));
    }
+
+   @Transactional
+   public List<BlobEntity> findBlobsByDirectory(String containerName, String directoryName, boolean recursive) {
+      ImmutableList.Builder<BlobEntity> result = ImmutableList.builder();
+      List<BlobEntity> blobEntities = blobRepository
+            .findBlobsByDirectory(containerRepository.findContainerByName(containerName), directoryName);
+      result.addAll(blobEntities);
+      if (recursive) {
+         for (BlobEntity blobEntity : blobEntities) {
+            if (blobEntity.isDirectory()) {
+               result.addAll(findBlobsByDirectory(containerName, blobEntity.getKey(), true));
+            }
+         }
+      }
+      return result.build();
+   }
+
+   @Transactional
+   public void deleteBlobsByContainer(String containerName) {
+      List<BlobEntity> blobs = findBlobsByContainer(containerName);
+      for (BlobEntity blob : blobs) {
+         deleteBlob(containerName, blob.getKey());
+      }
+   }
+
+   @Transactional
+   public void deleteBlobsByDirectory(String containerName, String directoryName, boolean recursive) {
+      List<BlobEntity> blobs = findBlobsByDirectory(containerName, directoryName, false);
+      for (BlobEntity blob : blobs) {
+         if (!blob.isDirectory()) {
+            deleteBlob(containerName, blob.getKey());
+         }
+         else {
+            if (recursive) {
+               deleteBlob(containerName, blob.getKey());
```

Comment:
Shouldn't this be deleted *after* all the blobs in the directory are deleted, after the recursive call?